tools/scripts/lock-api-state.sh

```diff
@@ -1,5 +1,11 @@
 #!/bin/bash -ex
 
+# check for dirty changes
```

Comment:
There's a check already in the Makefile, I just... commented it out at some point, I guess. (Also, I think this condition 
is reversed?)

Reply:
uhhh.. what makefile? I couldn't find anything calling this script?

Reply:
In `proto/Makefile` under `gen-buf-image` (which this calls, rather than the other way around).

Reply:
oh I see, it doesn't call the script, but the gen step has the check

Reply:
I will remove from the make file and leave it here. I remember it being annoying in the make file - when you want to make a breaking change, you know you have uncommitted changes, but when you are cutting a release, you want the check.

Reply:
look good to you @dzhu ?

Reply:
Makes sense!